shared/login/register/code-page/index.render.native.js

```diff
@@ -66,47 +57,54 @@ export default class CodePageRender extends Component {
     }
 
     if (!controls) {
-      return null
+      return <Box/>
     }
 
     return (
-      <View style={{flexDirection: 'column', justifyContent: 'space-between'}}>
-        {[
-          (<View style={{flexDirection: 'row', justifyContent: 'space-between', padding: 20, backgroundColor: 'orange'}}>
-            {controls.map(c => this.renderSwitch(c))}
-          </View>),
-          this.renderCameraBrokenControl()
-        ]}
-      </View>
+      <Box style={stylesHeader}>
+        <TabBar underlined tabWidth={150}>
+          {controls.map(c => this.renderControl(c))}
+        </TabBar>
+      </Box>
     )
   }
 
-  renderScanner () {
-    return (
-      <QR
-        scanning
-        onBarCodeRead={code => this.props.qrScanned(code)}
-        qrCode={this.props.qrCode}>
+  renderChangeMode () {
+    switch (this.props.myDeviceRole + this.props.otherDeviceRole) {
+      case codePageDeviceRoleNewPhone + codePageDeviceRoleExistingPhone:
+        if (this.props.mode === codePageModeScanCode) {
```

Comment:
this could be something like
```
const instructions = this.props.mode === codePageModeScanCode ? 'Type text code instead' : 'Scan QR code instead'
<Box style={stylesFooter}><Text>{instructions}</Text></Box>
```